src/warc2zim/converter.py

```diff
@@ -319,6 +321,22 @@ def gather_information_from_warc(self):
 
             # if we get here, found record for the main page
 
+            # if main page is a redirect, update the main url accordingly
+            if record.http_headers:
+                status_code = int(record.http_headers.get_statuscode())
+                if status_code in [
+                    HTTPStatus.MOVED_PERMANENTLY,
+                    HTTPStatus.FOUND,
+                ]:
+                    redirected_url = record.http_headers.get_header("Location")
+                    self.main_url_scheme = urlsplit(redirected_url).scheme
+                    self.main_url = normalize(redirected_url)
```

Comment:
If it doesn't include the scheme, this is misleading. I don't think we need this. Is this only to handle a special case for favicon? Then it shouldn't be there

Reply:
`main_url` is normalized since the beginning of zimit2 I think, and normalization removes the scheme.

`main_url_scheme` is for now only used for favicon, but we might need it for other things I think.

If it shouldn't be there, then where would you like to have it? If we are redirected from http to https on the main page, we probably need to know it, isn't it?

Reply:
OK, it's a naming issue then. We can't keep a `redirected_url` that's a regular URL and a `main_url` that's a scheme-less normalized URL next to each other and expect that people know those are different.

I think it would make sense to use `normalized_main_url` for the normalized version but if it's all over the place, we could do the other way around.
keeping normalized and scheme separately and having a property returning the main_url is fine as well.

Can you look into what would be the most practical?

Reply:
I've moved another way, computing the scheme only when needed, i.e. in `gather_information_from_warc` ; we do not need this information elsewhere in fact